tailcfg/c2ntypes.go

```diff
@@ -34,6 +34,15 @@ type C2NSSHUsernamesResponse struct {
 	Usernames []string
 }
 
+// C2NUpdateRequest is the request (from control to node) to the /update
+// handler. It tells the node what the latest version is. The node may choose
+// to ignore this update, notify the user, or apply it automatically (for POST
```

Comment:
Based on what? Configuration?

Should the node double check that 1.47.9384932849238493 isn't some bespoke version meant just for that node and not listed in the usual indexes? I'd call out how trustworthy this is.

Can this be used to downgrade? In the future? Or will that be a separate bool/field?

Reply:
Based on configuration, yes (see https://github.com/tailscale/tailscale/pull/8861).
Good point on validation. I've added some more of it in `handleC2NUpdate` and documented here that the value should not be blindly trusted.

> Should the node double check that 1.47.9384932849238493 isn't some bespoke version

Hmm, yes. We could check it against the `pkgs` JSON response or just avoid passing the version to `tailscale update` (let it figure out the latest version). I'm not sure if there's a better validation strategy, do you have any suggestions?

> Can this be used to downgrade?

I'd really really like to not support downgrades.
But if there's a good reason for it, we can add a bool to this request object and corresponding client validation changes.